<!DOCTYPE html><html lang="zh-CN" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0"><title>SMR 分析 | 方舟博客</title><meta name="author" content="方舟"><meta name="copyright" content="方舟"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="SMR 教程官方文档：SMR | Yang Lab (westlake.edu.cn) 1.数据获取1.1eqtl数据qtl数据获取主要从GTEx数据库，可以得到各种组织的数据。GTEx数据库的速度很慢建议从SMR官网的数据库下载（华人做的）SMR | Yang Lab (westlake.edu.cn)里面有各种qtl的数据集，下得很快！ eqtl还有血液的数据，样本量很大eQTLGen - c">
<meta property="og:type" content="article">
<meta property="og:title" content="SMR 分析">
<meta property="og:url" content="https://fangzhoufan.github.io/2024/06/14/10.SMR/index.html">
<meta property="og:site_name" content="方舟博客">
<meta property="og:description" content="SMR 教程官方文档：SMR | Yang Lab (westlake.edu.cn) 1.数据获取1.1eqtl数据qtl数据获取主要从GTEx数据库，可以得到各种组织的数据。GTEx数据库的速度很慢建议从SMR官网的数据库下载（华人做的）SMR | Yang Lab (westlake.edu.cn)里面有各种qtl的数据集，下得很快！ eqtl还有血液的数据，样本量很大eQTLGen - c">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://fangzhoufan.github.io/img/top_imge.jpg">
<meta property="article:published_time" content="2024-06-14T14:00:00.000Z">
<meta property="article:modified_time" content="2024-06-14T13:51:19.947Z">
<meta property="article:author" content="方舟">
<meta property="article:tag" content="孟德尔随机化,SMR">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://fangzhoufan.github.io/img/top_imge.jpg"><link rel="shortcut icon" href="https://picsum.photos/id/74/4288/2848"><link rel="canonical" href="https://fangzhoufan.github.io/2024/06/14/10.SMR/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = { 
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":200},
  copy: {
    success: '复制成功',
    error: '复制错误',
    noSupport: '浏览器不支持'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  date_suffix: {
    just: '刚刚',
    min: '分钟前',
    hour: '小时前',
    day: '天前',
    month: '个月前'
  },
  copyright: {"limitCount":50,"languages":{"author":"作者: 方舟","link":"链接: ","source":"来源: 方舟博客","info":"著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。"}},
  lightbox: 'fancybox',
  Snackbar: undefined,
  source: {
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/flickr-justified-gallery/dist/fjGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/flickr-justified-gallery/dist/fjGallery.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  }
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: 'SMR 分析',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2024-06-14 21:51:19'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(win=>{
    win.saveToLocal = {
      set: function setWithExpiry(key, value, ttl) {
        if (ttl === 0) return
        const now = new Date()
        const expiryDay = ttl * 86400000
        const item = {
          value: value,
          expiry: now.getTime() + expiryDay,
        }
        localStorage.setItem(key, JSON.stringify(item))
      },

      get: function getWithExpiry(key) {
        const itemStr = localStorage.getItem(key)

        if (!itemStr) {
          return undefined
        }
        const item = JSON.parse(itemStr)
        const now = new Date()

        if (now.getTime() > item.expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return item.value
      }
    }
  
    win.getScript = url => new Promise((resolve, reject) => {
      const script = document.createElement('script')
      script.src = url
      script.async = true
      script.onerror = reject
      script.onload = script.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        script.onload = script.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(script)
    })
  
    win.getCSS = (url,id = false) => new Promise((resolve, reject) => {
      const link = document.createElement('link')
      link.rel = 'stylesheet'
      link.href = url
      if (id) link.id = id
      link.onerror = reject
      link.onload = link.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        link.onload = link.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(link)
    })
  
      win.activateDarkMode = function () {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = function () {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
          if (t === 'dark') activateDarkMode()
          else if (t === 'light') activateLightMode()
        
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
    const detectApple = () => {
      if(/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
    })(window)</script><meta name="generator" content="Hexo 6.3.0"></head><body><div id="web_bg"></div><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="/./img/top_imge.jpg" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="sidebar-site-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">3</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">0</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">1</div></a></div><hr/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 首頁</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 時間軸</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 標籤</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分類</span></a></div><div class="menus_item"><a class="site-page group" href="javascript:void(0);"><i class="fa-fw fa fa-heartbeat"></i><span> 清單</span><i class="fas fa-chevron-down"></i></a><ul class="menus_item_child"><li><a class="site-page child" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友鏈</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 關於</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header"><nav id="nav"><span id="blog-info"><a href="/" title="方舟博客"><span class="site-name">方舟博客</span></a></span><div id="menus"><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> 首頁</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> 時間軸</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> 標籤</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> 分類</span></a></div><div class="menus_item"><a class="site-page group" href="javascript:void(0);"><i class="fa-fw fa fa-heartbeat"></i><span> 清單</span><i class="fas fa-chevron-down"></i></a><ul class="menus_item_child"><li><a class="site-page child" href="/Gallery/"><i class="fa-fw fas fa-images"></i><span> 照片</span></a></li></ul></div><div class="menus_item"><a class="site-page" href="/link/"><i class="fa-fw fas fa-link"></i><span> 友鏈</span></a></div><div class="menus_item"><a class="site-page" href="/about/"><i class="fa-fw fas fa-heart"></i><span> 關於</span></a></div></div><div id="toggle-menu"><a class="site-page" href="javascript:void(0);"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">SMR 分析</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">发表于</span><time class="post-meta-date-created" datetime="2024-06-14T14:00:00.000Z" title="发表于 2024-06-14 22:00:00">2024-06-14</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">更新于</span><time class="post-meta-date-updated" datetime="2024-06-14T13:51:19.947Z" title="更新于 2024-06-14 21:51:19">2024-06-14</time></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="SMR 分析"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">阅读量:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1 id="SMR-教程"><a href="#SMR-教程" class="headerlink" title="SMR 教程"></a>SMR 教程</h1><p>官方文档：<a target="_blank" rel="noopener" href="https://yanglab.westlake.edu.cn/software/smr/">SMR | Yang Lab (westlake.edu.cn)</a></p>
<h2 id="1-数据获取"><a href="#1-数据获取" class="headerlink" title="1.数据获取"></a>1.数据获取</h2><h3 id="1-1eqtl数据"><a href="#1-1eqtl数据" class="headerlink" title="1.1eqtl数据"></a>1.1eqtl数据</h3><p>qtl数据获取主要从GTEx数据库，可以得到各种组织的数据。GTEx数据库的速度很慢建议从SMR官网的数据库下载（华人做的）<a target="_blank" rel="noopener" href="https://yanglab.westlake.edu.cn/software/smr/#DataResource">SMR | Yang Lab (westlake.edu.cn)</a>里面有各种qtl的数据集，下得很快！</p>
<p>eqtl还有血液的数据，样本量很大<a target="_blank" rel="noopener" href="https://www.eqtlgen.org/cis-eqtls.html">eQTLGen - cis-eQTLs</a></p>
<p>直接下载besd二进制格式即可，后期数据可以用smr软件自提</p>
<h3 id="1-2GWAS数据"><a href="#1-2GWAS数据" class="headerlink" title="1.2GWAS数据"></a>1.2GWAS数据</h3><p>GWAS数据获取的方法很多可以直接去<a target="_blank" rel="noopener" href="https://gwas.mrcieu.ac.uk/">IEU OpenGWAS project (mrcieu.ac.uk)</a>下载，也可以通过R包TwoSampleMR直接从网站上拽。available_outcomes()函数可以看到openGWAS里面所有的gwas数据</p>
<p>当然还有很多别的获取手段。也可以从finngen的GWAS获取</p>
<h3 id="1-3参考基因组"><a href="#1-3参考基因组" class="headerlink" title="1.3参考基因组"></a>1.3参考基因组</h3><p>用R包下载<a target="_blank" rel="noopener" href="https://privefl.github.io/bigsnpr/reference/download_1000G.html">Download 1000G — download_1000G • bigsnpr (privefl.github.io)</a></p>
<p>也可以直接链接：<a target="_blank" rel="noopener" href="http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz">http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz</a></p>
<p>也可以自行从官网下载</p>
<p>不难发现，所需数据全是公共数据，而且获取比较轻松。</p>
<h2 id="2-SMR"><a href="#2-SMR" class="headerlink" title="2.SMR"></a>2.SMR</h2><h3 id="2-1软件下载"><a href="#2-1软件下载" class="headerlink" title="2.1软件下载"></a>2.1软件下载</h3><p><a target="_blank" rel="noopener" href="https://yanglab.westlake.edu.cn/software/smr/#Download">SMR | Yang Lab (westlake.edu.cn)</a>下载即可，Windows和Linux均可</p>
<figure class="highlight text"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">smr --bld mybld --gwas-summary mygwas.ma --beqtl-summary myeqtl --out mysmr --thread-num 10 </span><br></pre></td></tr></table></figure>

<p>smr输入有 </p>
<p>​	–bld 指的是1000人参考基因组的目录，</p>
<p>​	–gwas-summary输入的是GWAS数据，</p>
<p>​	–beqtl-summary 输入的是eqtl数据， </p>
<p>​	–query 5e-8 </p>
<p>​	–gene ENSG00000113161 </p>
<p>​	–out 输出文集名</p>
<p>​	 –make-besd</p>
<p>*<strong>mygwas.ma*</strong></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">SNP    A1  A2  freq    b   se  p   n</span><br><span class="line">rs1001    A   G   0.8493  0.0024  0.0055  0.6653  129850</span><br><span class="line">rs1002    C   G   0.03606 0.0034  0.0115  0.7659  129799</span><br><span class="line">rs1003    A   C   0.5128  0.045   0.038   0.2319  129830</span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">exp</span><span class="operator">&lt;-</span>fread<span class="punctuation">(</span><span class="string">&#x27;GCST004744-EFO_0000571-Build37.f.tsv.gz&#x27;</span><span class="punctuation">,</span>data.table <span class="operator">=</span> <span class="built_in">F</span><span class="punctuation">,</span>header<span class="operator">=</span><span class="built_in">T</span><span class="punctuation">)</span></span><br><span class="line">tail<span class="punctuation">(</span><span class="built_in">exp</span><span class="punctuation">)</span></span><br><span class="line"><span class="built_in">exp</span><span class="operator">$</span>n<span class="operator">&lt;-</span><span class="number">66756</span></span><br><span class="line"><span class="comment">##ebi常用格式</span></span><br><span class="line">LUAD<span class="operator">&lt;-</span><span class="built_in">exp</span><span class="operator">%&gt;%</span></span><br><span class="line">  dplyr<span class="operator">::</span>filter<span class="punctuation">(</span>str_detect<span class="punctuation">(</span>variant_id<span class="punctuation">,</span> <span class="string">&quot;^rs&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="operator">%&gt;%</span></span><br><span class="line">  dplyr<span class="operator">::</span>rename<span class="punctuation">(</span></span><br><span class="line">    SNP<span class="operator">=</span>variant_id<span class="punctuation">,</span></span><br><span class="line">    A1<span class="operator">=</span>effect_allele<span class="punctuation">,</span> </span><br><span class="line">    A2<span class="operator">=</span>other_allele<span class="punctuation">,</span></span><br><span class="line">    freq<span class="operator">=</span>effect_allele_frequency<span class="punctuation">,</span> </span><br><span class="line">    b<span class="operator">=</span>beta<span class="punctuation">,</span></span><br><span class="line">    se<span class="operator">=</span>standard_error<span class="punctuation">,</span></span><br><span class="line">    p<span class="operator">=</span>p_value<span class="punctuation">,</span></span><br><span class="line">    N<span class="operator">=</span>n </span><br><span class="line">  <span class="punctuation">)</span><span class="operator">%&gt;%</span></span><br><span class="line">  dplyr<span class="operator">::</span>select<span class="punctuation">(</span>SNP<span class="punctuation">,</span> A1<span class="punctuation">,</span> A2<span class="punctuation">,</span> freq<span class="punctuation">,</span> b<span class="punctuation">,</span> se<span class="punctuation">,</span> p<span class="punctuation">,</span> N<span class="punctuation">)</span> <span class="operator">%&gt;%</span></span><br><span class="line">  dplyr<span class="operator">::</span>filter<span class="punctuation">(</span><span class="operator">!</span>str_detect<span class="punctuation">(</span>SNP<span class="punctuation">,</span> <span class="string">&quot;,&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment">##finngen常用格式</span></span><br><span class="line">LUAD<span class="operator">&lt;-</span><span class="built_in">exp</span><span class="operator">%&gt;%</span></span><br><span class="line">  dplyr<span class="operator">::</span>filter<span class="punctuation">(</span>str_detect<span class="punctuation">(</span>rsids<span class="punctuation">,</span> <span class="string">&quot;^rs&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span><span class="operator">%&gt;%</span></span><br><span class="line">  dplyr<span class="operator">::</span>rename<span class="punctuation">(</span></span><br><span class="line">    SNP<span class="operator">=</span>rsids<span class="punctuation">,</span></span><br><span class="line">    A1<span class="operator">=</span>alt<span class="punctuation">,</span> </span><br><span class="line">    A2<span class="operator">=</span>ref<span class="punctuation">,</span></span><br><span class="line">    freq<span class="operator">=</span>af_alt<span class="punctuation">,</span> </span><br><span class="line">    b<span class="operator">=</span>beta<span class="punctuation">,</span></span><br><span class="line">    se<span class="operator">=</span>sebeta<span class="punctuation">,</span></span><br><span class="line">    p<span class="operator">=</span>pval<span class="punctuation">,</span></span><br><span class="line">    N<span class="operator">=</span>n </span><br><span class="line">  <span class="punctuation">)</span><span class="operator">%&gt;%</span></span><br><span class="line">  dplyr<span class="operator">::</span>select<span class="punctuation">(</span>SNP<span class="punctuation">,</span> A1<span class="punctuation">,</span> A2<span class="punctuation">,</span> freq<span class="punctuation">,</span> b<span class="punctuation">,</span> se<span class="punctuation">,</span> p<span class="punctuation">,</span> N<span class="punctuation">)</span> <span class="operator">%&gt;%</span></span><br><span class="line">  dplyr<span class="operator">::</span>filter<span class="punctuation">(</span><span class="operator">!</span>str_detect<span class="punctuation">(</span>SNP<span class="punctuation">,</span> <span class="string">&quot;,&quot;</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>LUAD<span class="punctuation">,</span><span class="number">804</span><span class="punctuation">)</span></span><br><span class="line">write.table<span class="punctuation">(</span>LUAD<span class="punctuation">,</span><span class="string">&#x27;LUAD.ma&#x27;</span><span class="punctuation">,</span> sep <span class="operator">=</span> <span class="string">&quot;\t&quot;</span><span class="punctuation">,</span> row.names <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">,</span> col.names <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">,</span> <span class="built_in">quote</span> <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>



<blockquote>
<p><a target="_blank" rel="noopener" href="https://yanglab.westlake.edu.cn/software/smr/#QueryeQTLSummaryResults">SMR 具体eqtlgen转换要求</a></p>
</blockquote>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">#超算平台</span><br><span class="line">cd &quot;/public/home/fanfangzhou/apps/SMR/smr_linux&quot; </span><br><span class="line"></span><br><span class="line">##这里获取的是含有eQTLGen的HMGCR这基因的besd格式的</span><br><span class="line">&quot;/public/home/fanfangzhou/apps/SMR/smr_linux&quot; </span><br><span class="line">	--beqtl-summary &quot;/public/home/fanfangzhou/R/GTEx8/SMR/eQTLGen/cis-eQTLs-full_eQTLGen_AF_incl_nr_formatted_20191212.new.txt_besd-dense&quot; </span><br><span class="line">	--query 1 </span><br><span class="line">	--gene ENSG00000113161 </span><br><span class="line">	--out &quot;/public/home/fanfangzhou/R/2024science/6. Drugs mendelian/rotator cuff syndrome/rotator cuff injury/SMR/&quot; </span><br><span class="line">	--make-besd</span><br><span class="line">	</span><br><span class="line">##这里获取了HMGCR的besd格式文件，就可以在这个区间进行SMR分析</span><br><span class="line">&quot;/public/home/fanfangzhou/apps/SMR/smr_linux&quot;  </span><br><span class="line">	--bfile &quot;/public/home/fanfangzhou/R/Rcodes/486metabolites/1kg.v3/EUR&quot; </span><br><span class="line">	--gwas-summary SMR_rotatorcuffR9.ma </span><br><span class="line">	--beqtl-summary HMGCR </span><br><span class="line">	--out rotatorblood  </span><br><span class="line">	--thread-num 10 </span><br></pre></td></tr></table></figure>



<h3 id="2-3运行smr"><a href="#2-3运行smr" class="headerlink" title="2.3运行smr"></a>2.3运行smr</h3><p>命令很简单，找着官网的改一改就行</p>
<figure class="highlight text"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">##超算平台</span><br><span class="line">&quot;/public/home/fanfangzhou/apps/SMR/smr_linux&quot;  --bfile &quot;/public/home/fanfangzhou/R/Rcodes/486metabolites/1kg.v3/EUR&quot; --gwas-summary LUAD.ma --beqtl-summary &quot;/public/home/fanfangzhou/R/GTEx8/SMR/Whole_Blood/Whole_Blood&quot; --out Lung_SMR_wholeblood --thread-num 10 --diff-freq-prop 0.92</span><br></pre></td></tr></table></figure>

<p>–<strong>diff-freq-prop 参数是等位基因频率的设定阈值，如果小的话程序是跑不通的。因为我输入的eqtl和gwas数据没经过过滤，被筛掉的snp的比例非常大，所以需要把这个阈值设高点防止程序终止。</strong></p>
<h2 id="3-共定位"><a href="#3-共定位" class="headerlink" title="3.共定位"></a>3.共定位</h2><p>细心的朋友会发现，刚才我们并没有下载eqtl的txt列表格式的文件，理论上和gwas一样，eqtl也应该有一个非常大数据量的列表文件，那么如何得到呢，依然是靠smr软件</p>
<p>这里可以用eqtlgen的也可以用V8的，***–make-besd<em><strong>加这个就是输出</strong>besd</em><em>格式的（三种文件那种），或者不加就是默认*<em>txt</em></em></p>
<figure class="highlight text"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">smr-1.3.1-win.exe --beqtl-summary Artery_Aorta --query 0.05 --out Aorta_eqtl</span><br></pre></td></tr></table></figure>

<p>smr可以从二进制的besd文件中导出txt格式的eqtl文件，得到这里是Aorta_eqtl.txt文件</p>
<h2 id="4-SMR-locus-plot"><a href="#4-SMR-locus-plot" class="headerlink" title="4.SMR locus plot"></a>4.SMR locus plot</h2><p><a target="_blank" rel="noopener" href="https://yanglab.westlake.edu.cn/software/smr/#SMRlocusplot19">SMR | Yang Lab (westlake.edu.cn)</a></p>
<h3 id="SMR-locus-plot"><a href="#SMR-locus-plot" class="headerlink" title="SMR locus plot"></a>SMR locus plot</h3><p>Here we provide an R script to plot SMR results as presented in Zhu et al. (<a target="_blank" rel="noopener" href="http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.3538.html">2016 Nature Genetics</a>). The data file for plot can be generated by the command below. The R script is avaliable at <a target="_blank" rel="noopener" href="https://yanglab.westlake.edu.cn/software/smr/#Download">Download</a>.</p>
<h4 id="SMR-command-line-to-generate-a-data-file-for-plot"><a href="#SMR-command-line-to-generate-a-data-file-for-plot" class="headerlink" title="# SMR command line to generate a data file for plot"></a># SMR command line to generate a data file for plot</h4><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">smr --bfile mydata --gwas-summary mygwas.ma --beqtl-summary myeqtl --out myplot --plot --probe ILMN_123 --probe-wind 500 --gene-list glist-hg19 </span><br></pre></td></tr></table></figure>



<h2 id="5-Multi-SNP-based-SMR-test"><a href="#5-Multi-SNP-based-SMR-test" class="headerlink" title="5.Multi-SNP-based SMR test"></a>5.Multi-SNP-based SMR test</h2><p>Below shows an option to combine the information from all the SNPs in a region that pass a p-value threshold (the default value is 5.0e-8 which can be modified by the flag <strong>–peqtl-smr</strong>) to conduct a multi-SNP-based SMR analysis (<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41467-018-03371-0">Wu et al. 2018 Nature Communications</a>).</p>
<p>The SNPs are pruned for LD using a weighted vertex coverage algorithm with a LD r2 threshold (the default value is 0.9 which can be modified by the flag <strong>–ld-pruning</strong>) and eQTL p-value as the weight.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">smr --bfile mydata --gwas-summary mygwas.ma --beqtl-summary myeqtl --out mymulti --smr-multi </span><br></pre></td></tr></table></figure>

<p><strong>–smr-multi</strong> turns on set-based SMR test in the cis-region.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">smr --bfile mydata --gwas-summary mygwas.ma --beqtl-summary myeqtl --out mymulti --smr-multi --set-wind 500 </span><br></pre></td></tr></table></figure>

<p><strong>–set-wind</strong> defines a window width (Kb) centred around the top associated cis-eQTL to select SNPs in the cis-region. The default value is -9 resulting in selecting SNPs in the whole cis-region if this option is not specified.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">smr --bfile mydata --gwas-summary mygwas.ma --beqtl-summary myeqtl --out mymulti --smr-multi  --ld-multi-snp 0.1 </span><br></pre></td></tr></table></figure>

<p><strong>–ld-multi-snp</strong> LD r-squared threshold used to prune SNPs (eQTLs) in the Multi-SNP based SMR test. The default value is 0.1.</p>
<blockquote>
<p>最初提出的定义独立顺式-eQTL的阈值为<em>r</em>2&lt; 0.9。然而，该阈值可能包括具有高 LD 的 SNP（定义为 <em>r</em> 2≥ 0.1）。因此，我们采用了 <em>r</em> 2&lt; 0.1 作为定义独立 SNP 的阈值</p>
</blockquote>
<p>参考文献：<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9538104/">基于转录组全汇总数据的孟德尔随机化分析揭示了 38 个与严重 COVID-19 相关的新基因 - PMC (nih.gov)</a></p>
<h3 id="GTEx-V8-基因注释"><a href="#GTEx-V8-基因注释" class="headerlink" title="GTEx V8 基因注释"></a>GTEx V8 基因注释</h3><p><a target="_blank" rel="noopener" href="https://gtexportal.org/home/downloads/adult-gtex/bulk_tissue_expression">GTEx Portal</a>网址里面的：<a target="_blank" rel="noopener" href="https://storage.googleapis.com/adult-gtex/bulk-gex/v8/rna-seq/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_reads.gct.gz">https://storage.googleapis.com/adult-gtex/bulk-gex/v8/rna-seq/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_reads.gct.gz</a></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">GTEx&lt;-read.table(&quot;GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_reads.gct&quot;, skip = 2, header = TRUE, sep = &quot;\t&quot;) save(GTEx,file = &#x27;GTEx.Rdata&#x27;) </span><br><span class="line">GTEx[1:4,1:4] ##行是基因 列是样本 </span><br><span class="line"># Name    Description       GTEX.1117F.0226.SM.5GZZ7 GTEX.1117F.0426.SM.5EGHI </span><br><span class="line"># 1 ENSG00000223972.5 DDX11L1 0 0 </span><br><span class="line"># 2 ENSG00000227232.5 WASH7P 187 109 </span><br><span class="line"># 3 ENSG00000278267.1 MIR6859-1 0 0 </span><br><span class="line"># 4 ENSG00000243485.5 MIR1302-2HG 1 0 </span><br><span class="line">colnames(GTEx)</span><br></pre></td></tr></table></figure>





<h2 id="5-常用数据"><a href="#5-常用数据" class="headerlink" title="5.常用数据"></a>5.常用数据</h2><h3 id="V8-release-of-the-GTEx-eQTL-x2F-sQTL-summary-data"><a href="#V8-release-of-the-GTEx-eQTL-x2F-sQTL-summary-data" class="headerlink" title="V8 release of the GTEx eQTL&#x2F;sQTL summary data"></a><em>V8 release of the GTEx eQTL&#x2F;sQTL summary data</em></h3><p>V8 release of the GTEx eQTL&#x2F;sQTL summary data (n &#x3D; 73 - 670; [GTEx Consortium 2020 Science](<a target="_blank" rel="noopener" href="https://www.science.org/doi/10.1126/science.aaz1776?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub">https://www.science.org/doi/10.1126/science.aaz1776?url_ver=Z39.88-2003&amp;rfr_id=ori:rid:crossref.org&amp;rfr_dat=cr_pub</a>  0pubmed)) in SMR binary (BESD) format:</p>
<p><a target="_blank" rel="noopener" href="https://yanglab.westlake.edu.cn/data/SMR/GTEx_V8_cis_eqtl_summary.html">GTEx_V8_cis_eqtl_summary(hg19)</a> (48 GB)</p>
<p><a target="_blank" rel="noopener" href="https://yanglab.westlake.edu.cn/data/SMR/GTEx_V8_cis_eqtl_summary_lite.tar">GTEx_V8_cis_eqtl_summary_lite</a> (hg19) Lite version of the GTEx V8 eqtl data (only SNPs with P &lt; 1e-5 are included; 506MB)</p>
<p>This is a set of cis-eQTL summary data across 49 human tissues from the GTEx project. Only SNPs within 1Mb of the transcription start site are available. The forth column of the *.epi file is the middle position of the probe sequence rather than the transcription start site.</p>
<h3 id="cis-eQTLs"><a href="#cis-eQTLs" class="headerlink" title="cis-eQTLs"></a><em>cis</em>-eQTLs</h3><p>This page contains the <em>cis</em>-eQTL results. The statistically significant <em>cis</em>-eQTLs and SMR-prioritised genes for several traits are browsable, the other files can be downloaded.</p>
<h5 id="Downloads"><a href="#Downloads" class="headerlink" title="Downloads"></a>Downloads</h5><ul>
<li><p><a target="_blank" rel="noopener" href="https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gz">Significant <em>cis</em>-eQTLs</a></p>
</li>
<li><p><a target="_blank" rel="noopener" href="https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/2019-12-11-cis-eQTLsFDR-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gz">Full <em>cis</em>-eQTL summary statistics</a></p>
</li>
<li><p><a target="_blank" rel="noopener" href="https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/README_cis">README</a></p>
</li>
<li><p><a target="_blank" rel="noopener" href="https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/cis-eQTL_significant_20181017.txt.gz">Initial release Significant <em>cis</em>-eQTLs</a> (deprecated)</p>
</li>
<li><p><a target="_blank" rel="noopener" href="https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/cis-eQTLs_full_20180905.txt.gz">Initial release Full <em>cis</em>-eQTL summary statistics</a> (deprecated)</p>
</li>
</ul>
<h2 id="6-SMR使用技巧"><a href="#6-SMR使用技巧" class="headerlink" title="6.SMR使用技巧"></a>6.SMR使用技巧</h2><h3 id="6-1-批量使用49个组织的SMR"><a href="#6-1-批量使用49个组织的SMR" class="headerlink" title="6.1.批量使用49个组织的SMR"></a>6.1.批量使用49个组织的SMR</h3><p>首先在R里面创建我们需要的sh格式的文件，当然我们可以进行txt简单创建</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line">original_string1 <span class="operator">&lt;-</span> <span class="string">&#x27;</span></span><br><span class="line"><span class="string">echo Whole_Blood</span></span><br><span class="line"><span class="string">&quot;/public/home/fanfangzhou/apps/SMR/smr_linux&quot; --bfile &quot;/public/home/fanfangzhou/R/Rcodes/486metabolites/1kg.v3/EUR&quot; --gwas-summary LUAD.ma --beqtl-summary &quot;/public/home/fanfangzhou/R/GTEx8/SMR/Whole_Blood/Whole_Blood&quot; --out Lung_SMR_Whole_Blood --thread-num 10 --diff-freq-prop 0.92</span></span><br><span class="line"><span class="string">&#x27;</span></span><br><span class="line">original_string2 <span class="operator">&lt;-</span> <span class="string">&#x27;</span></span><br><span class="line"><span class="string">echo Whole_Blood</span></span><br><span class="line"><span class="string">&quot;/public/home/fanfangzhou/apps/SMR/smr_linux&quot; --bfile &quot;/public/home/fanfangzhou/R/Rcodes/486metabolites/1kg.v3/EUR&quot; --gwas-summary LUAD.ma --beqtl-summary &quot;/public/home/fanfangzhou/R/GTEx8/SMR/Whole_Blood/Whole_Blood&quot; --out Lung_SMR_Whole_Blood_test  --smr-multi  --ld-multi-snp 0.1 </span></span><br><span class="line"><span class="string">&#x27;</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 待替换的字符串</span></span><br><span class="line">replace_values <span class="operator">&lt;-</span> original_string1  <span class="comment"># 替换值的向量</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 用替换值逐个替换原始字符串中的&quot;Whole_Blood&quot;</span></span><br><span class="line">replaced_strings <span class="operator">&lt;-</span> sapply<span class="punctuation">(</span>replace_values<span class="punctuation">,</span> <span class="keyword">function</span><span class="punctuation">(</span>value<span class="punctuation">)</span> <span class="punctuation">&#123;</span></span><br><span class="line">  gsub<span class="punctuation">(</span><span class="string">&quot;Whole_Blood&quot;</span><span class="punctuation">,</span> value<span class="punctuation">,</span> original_string1<span class="punctuation">)</span></span><br><span class="line"><span class="punctuation">&#125;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">writeLines<span class="punctuation">(</span>replaced_strings<span class="punctuation">,</span> <span class="string">&quot;script.sh&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">command <span class="operator">&lt;-</span> <span class="string">&#x27;</span></span><br><span class="line"><span class="string">#!/bin/bash</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">ls</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string"># 输出 &quot;Hello, world!&quot; 消息</span></span><br><span class="line"><span class="string">echo &quot;Hello, world!&quot;</span></span><br><span class="line"><span class="string">&#x27;</span></span><br><span class="line">writeLines<span class="punctuation">(</span>command<span class="punctuation">,</span> <span class="string">&quot;script.sh&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<p>然后我们在linux系统里面使用sh格式运行</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#使用 chmod 命令将该文件设置为可执行权限：</span></span><br><span class="line"><span class="built_in">chmod</span> +x my_script.sh</span><br><span class="line"></span><br><span class="line"><span class="comment">#执行该脚本文件：</span></span><br><span class="line">./my_script.sh</span><br></pre></td></tr></table></figure>

<h3 id="6-2-SMR-执行mult"><a href="#6-2-SMR-执行mult" class="headerlink" title="6.2.SMR 执行mult"></a>6.2.SMR 执行mult</h3><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">&quot;/public/home/fanfangzhou/apps/SMR/smr_linux&quot; --bfile &quot;/public/home/fanfangzhou/R/Rcodes/486metabolites/1kg.v3/EUR&quot; --gwas-summary ON.ma --beqtl-summary &quot;/public/home/fanfangzhou/R/GTEx8/SMR/eQTLGen/cis-eQTLs-full_eQTLGen_AF_incl_nr_formatted_20191212.new.txt_besd-dense&quot;  --out osteonecrosis_blood  --smr-multi  --ld-multi-snp 0.1 </span><br></pre></td></tr></table></figure>

</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">文章作者: </span><span class="post-copyright-info"><a href="https://fangzhoufan.github.io">方舟</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">文章链接: </span><span class="post-copyright-info"><a href="https://fangzhoufan.github.io/2024/06/14/10.SMR/">https://fangzhoufan.github.io/2024/06/14/10.SMR/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">版权声明: </span><span class="post-copyright-info">本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0</a> 许可协议。转载请注明来自 <a href="https://fangzhoufan.github.io" target="_blank">方舟博客</a>！</span></div></div><div class="tag_share"><div class="post-meta__tag-list"></div><div class="post_share"><div class="social-share" data-image="/./img/top_imge.jpg" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="next-post pull-full"><a href="/2024/06/14/2.TWAS%20&amp;%20FUSION/" title="TWAS 分析"><img class="cover" src="http://gusevlab.org/projects/fusion/SCZ_chr22.png" onerror="onerror=null;src='/img/404.jpg'" alt="cover of next post"><div class="pagination-info"><div class="label">下一篇</div><div class="next_info">TWAS 分析</div></div></a></div></nav></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="is-center"><div class="avatar-img"><img src="/./img/top_imge.jpg" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info__name">方舟</div><div class="author-info__description">要人心之自由，胸襟开放；要有一个远见，超越未见</div></div><div class="card-info-data site-data is-center"><a href="/archives/"><div class="headline">文章</div><div class="length-num">3</div></a><a href="/tags/"><div class="headline">标签</div><div class="length-num">0</div></a><a href="/categories/"><div class="headline">分类</div><div class="length-num">1</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/fangzhoufan"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://github.com/fangzhoufan" target="_blank" title="Github"><i class="fab fa-github"></i></a><a class="social-icon" href="mailto:haohaojiu@outlook.com" target="_blank" title="Email"><i class="fas fa-envelope"></i></a><a class="social-icon" href="https://www.zhihu.com/people/ha-hao-jiu-54" target="_blank" title="Zhihu"><i class="fab fa-zhihu"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>公告</span></div><div class="announcement_content">This is my Blog</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>目录</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#SMR-%E6%95%99%E7%A8%8B"><span class="toc-number">1.</span> <span class="toc-text">SMR 教程</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-%E6%95%B0%E6%8D%AE%E8%8E%B7%E5%8F%96"><span class="toc-number">1.1.</span> <span class="toc-text">1.数据获取</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-1eqtl%E6%95%B0%E6%8D%AE"><span class="toc-number">1.1.1.</span> <span class="toc-text">1.1eqtl数据</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#1-2GWAS%E6%95%B0%E6%8D%AE"><span class="toc-number">1.1.2.</span> <span class="toc-text">1.2GWAS数据</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#1-3%E5%8F%82%E8%80%83%E5%9F%BA%E5%9B%A0%E7%BB%84"><span class="toc-number">1.1.3.</span> <span class="toc-text">1.3参考基因组</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-SMR"><span class="toc-number">1.2.</span> <span class="toc-text">2.SMR</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#2-1%E8%BD%AF%E4%BB%B6%E4%B8%8B%E8%BD%BD"><span class="toc-number">1.2.1.</span> <span class="toc-text">2.1软件下载</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-3%E8%BF%90%E8%A1%8Csmr"><span class="toc-number">1.2.2.</span> <span class="toc-text">2.3运行smr</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-%E5%85%B1%E5%AE%9A%E4%BD%8D"><span class="toc-number">1.3.</span> <span class="toc-text">3.共定位</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#4-SMR-locus-plot"><span class="toc-number">1.4.</span> <span class="toc-text">4.SMR locus plot</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#SMR-locus-plot"><span class="toc-number">1.4.1.</span> <span class="toc-text">SMR locus plot</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#SMR-command-line-to-generate-a-data-file-for-plot"><span class="toc-number">1.4.1.1.</span> <span class="toc-text"># SMR command line to generate a data file for plot</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-Multi-SNP-based-SMR-test"><span class="toc-number">1.5.</span> <span class="toc-text">5.Multi-SNP-based SMR test</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#GTEx-V8-%E5%9F%BA%E5%9B%A0%E6%B3%A8%E9%87%8A"><span class="toc-number">1.5.1.</span> <span class="toc-text">GTEx V8 基因注释</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#5-%E5%B8%B8%E7%94%A8%E6%95%B0%E6%8D%AE"><span class="toc-number">1.6.</span> <span class="toc-text">5.常用数据</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#V8-release-of-the-GTEx-eQTL-x2F-sQTL-summary-data"><span class="toc-number">1.6.1.</span> <span class="toc-text">V8 release of the GTEx eQTL&#x2F;sQTL summary data</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#cis-eQTLs"><span class="toc-number">1.6.2.</span> <span class="toc-text">cis-eQTLs</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#Downloads"><span class="toc-number">1.6.2.0.1.</span> <span class="toc-text">Downloads</span></a></li></ol></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#6-SMR%E4%BD%BF%E7%94%A8%E6%8A%80%E5%B7%A7"><span class="toc-number">1.7.</span> <span class="toc-text">6.SMR使用技巧</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#6-1-%E6%89%B9%E9%87%8F%E4%BD%BF%E7%94%A849%E4%B8%AA%E7%BB%84%E7%BB%87%E7%9A%84SMR"><span class="toc-number">1.7.1.</span> <span class="toc-text">6.1.批量使用49个组织的SMR</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#6-2-SMR-%E6%89%A7%E8%A1%8Cmult"><span class="toc-number">1.7.2.</span> <span class="toc-text">6.2.SMR 执行mult</span></a></li></ol></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>最新文章</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/06/14/10.SMR/" title="SMR 分析">SMR 分析</a><time datetime="2024-06-14T14:00:00.000Z" title="发表于 2024-06-14 22:00:00">2024-06-14</time></div></div><div class="aside-list-item"><a class="thumbnail" href="/2024/06/14/2.TWAS%20&amp;%20FUSION/" title="TWAS 分析"><img src="http://gusevlab.org/projects/fusion/SCZ_chr22.png" onerror="this.onerror=null;this.src='/img/404.jpg'" alt="TWAS 分析"/></a><div class="content"><a class="title" href="/2024/06/14/2.TWAS%20&amp;%20FUSION/" title="TWAS 分析">TWAS 分析</a><time datetime="2024-06-14T14:00:00.000Z" title="发表于 2024-06-14 22:00:00">2024-06-14</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/06/14/4.LDSC%20software/" title="LDSC 分析">LDSC 分析</a><time datetime="2024-06-14T14:00:00.000Z" title="发表于 2024-06-14 22:00:00">2024-06-14</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2024 By 方舟</div><div class="framework-info"><span>框架 </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>主题 </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="阅读模式"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="浅色和深色模式转换"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="单栏和双栏切换"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="设置"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="目录"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="回到顶部"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox.umd.min.js"></script><div class="js-pjax"></div><script id="click-heart" src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/dist/click-heart.min.js" async="async" mobile="false"></script><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>